packages/core/src/drawer/drawer.ts

```diff
@@ -0,0 +1,110 @@
+import type { TemplateResult } from 'lit-html';
+import type { CSSResultGroup } from 'lit';
+import type { ButtonSize } from '../button/button.js';
+import { LitElement, html } from 'lit';
+import { property, query } from 'lit/decorators.js';
+import styles from './drawer.scss.js';
+
+export type DrawerAttachement = 'right' | 'left' | 'top' | 'bottom';
+
+/**
+ * A dialog component for displaying modal UI.
+ *
+ * @slot default - Body content for the dialog
+ * @slot header - Header content for the dialog
+ * @slot title - The title of the dialog
+ */
+export class Drawer extends LitElement {
+  /** @private */
+  static override styles: CSSResultGroup = styles;
+
+  @query('dialog') dialog?: HTMLDialogElement;
+
+  /** Disables the ability to close the dialog using the Escape key. */
+  @property({ type: Boolean, attribute: 'disable-close' }) disableClose = false;
+
+  /** The ARIA role of the dialog. */
+  @property() override role: 'dialog' | 'section' = 'dialog';
+
+  /** The side of the screen where the drawer is attached */
+  @property({ reflect: true }) attachment: DrawerAttachement = 'right';
+
+  /** The size of the button */
+  @property() closeButtonSize: ButtonSize = 'sm';
+
+  override connectedCallback(): void {
+    super.connectedCallback();
+
+    this.inert = true;
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <dialog
+        @cancel=${this.#onCancel}
+        @click=${this.#onClick}
+        @close=${this.#onClose}
+        .role=${this.role}
+        aria-labelledby="title"
+        part="dialog"
+      >
+        <div>
+          <sl-button sl-dialog-close .size="${this.closeButtonSize}" aria-label="back to page" title="close"
+            >x</sl-button
+          >
+          <slot name="title" id="title"></slot>
+          <slot name="actions"></slot>
```

Comment:
So in DNA, I usually wrap the `<slot name="actions">` in a button-bar (`<sl-button-bar>`); that way the button-bar automatically spaces the buttons if they are next to each other.